map.js

```diff
@@ -304,6 +305,31 @@ $(function() {
         return false
     })
 
+    window.addEventListener("hashchange", function(){
+        openStudioPopup(location.hash.substr(1));
+    })
+
+
+    function openStudioPopup(studioId){
+        for (var b in studios) {
+            if (studios[b].id == studioId) {
+                var studioCoordinates = studios[b].coords
+                mymap.setView([studioCoordinates[1], studioCoordinates[0]], 14)
+                break
+            }
+        }
+
+        setTimeout(function() {
+            for (var i in studiosMarkers) {
```

Comment:
I don't think that .filter will be better for this. In my cycle when the condition is met, we will exit the loop. Filter will work with all the array, no matter how long it will be, even if it meet the condition. 

Maybe I didn't understand it right.